tests/modules/services/borgmatic/basic-configuration.nix

```diff
@@ -0,0 +1,40 @@
+{ config, pkgs, ... }:
+
+let
+  source_directory = "/my-stuff-to-backup";
+  repository = "/my-backup-disk";
+in {
+  config = {
+    services.borgmatic = {
+      enable = true;
+      configs = {
+        main = {
+          location = {
+            source_directories = [ source_directory ];
+            repositories = [ repository ];
+            extraConfig = { one_file_system = true; };
+          };
+        };
+      };
+    };
+
+    test.stubs.borgmatic = { };
+
+    nmt.script = ''
+      config_file=$TESTED/home-files/.config/borgmatic.d/main.yaml
+      assertFileExists $config_file
+
+      yq=${pkgs.yq-go}/bin/yq
+
+      actual_source_directory=$($yq ".location.source_directories[0]" $config_file)
+      if [[ $actual_source_directory != ${source_directory} ]]; then
+         fail "Expected source directory to be ${source_directory} but was $actual_source_directory"
+      fi
+
+      one_file_system=$($yq ".location.one_file_system" $config_file)
+      if [[ $one_file_system != "true" ]]; then
+         fail "Expected one_file_system to be true but it was $one_file_system"
+      fi
```

Comment:
Perhaps it would be possible to simply use `assertFileContent` to verify the entire generated file?

Reply:
Could also assert the content of the generated service and timer files…

Reply:
> Perhaps it would be possible to simply use assertFileContent to verify the entire generated file?

I will do it if you really want me to but I would prefer not because it would make the test more fragile, i.e., the test would fail for unrelated reasons. E.g.,
- the test would fail if `builtins.toYAML` starts sorting keys of the input attribute set;
- the test would fail if `builtins.toYAML` starts output a file that looks like typical YAML instead of [outputting JSON](https://github.com/NixOS/nixpkgs/blob/master/lib/generators.nix#L243) as it does today

By using a YAML parser and checking attributes one by one, I make sure borgmatic will be happy with the configuration file without checking a particular implementation detail of `builtins.toYAML`.

Instead, I added checks for much more options. I hope it is ok with you. If you still prefer me to use `assertFileContent`, I will.

> Could also assert the content of the generated service and timer files…

done. I used `assertFileContent` here.